### DISCUSSION TITLE
PostSync support?

### DISCUSSION THREAD

**Discussion by AshleyDumaine** [fluxcd/flux2#1183] [Q&A] [answered]
Hey, I was looking at migrating to Flux v2 or ArgoCD (currently on Flux v1 now) and was wondering if there's something similar to [ArgoCD's PostSync resource hook](https://argoproj.github.io/argo-cd/user-guide/resource_hooks/)? I'd want to automatically run a script/curl command after the sync completes successfully. Thanks!

**Comment by phillebaba** [comment, marked as answer]
I would suggest you look into the notification controller and what is possible to do with it. There is a generic webhook notifier if all you want is to post the event to some other system. Is your plan to simply make a http request somewhere or is it to run more complex commands?
https://toolkit.fluxcd.io/guides/notifications/#setup-notifications

**Reply by AshleyDumaine** [reply]
More complex commands are needed rather than just posting the event json to another system. The post-sync even I was thinking of would be an interaction with the k8s API to obtain Service `externalIP`s plus API calls to a cluster-external system with the k8s API results.

**Reply by phillebaba** [reply]
So these types of post hook commands is not a feature in Flux right now. Out of interest how does ArgoCD implement these this? Is there a separate container deployed which runs the command or is it run inside of the ArgoCD container?

I dont know how much time you are willing to spend on the issue, but @bigkevmcd has written a blog post about how to trigger tekton pipelines after a deployment. It would solve your problem but it will require you to also deploy Tekton.
https://bigkevmcd.github.io/tekton/flux/notifications/2020/12/06/flux-tekton.html

I am not voting against implementing this in Flux but I think it would be better to make the integration with Tekton easier, as we would gain all the great features Tekton has implemented.

**Reply by AshleyDumaine** [reply]
I haven't been able to experiment with ArgoCD yet, but plan on doing so within the next couple months. From the docs, it looks like these PostSync actions can be executed via [K8s Jobs](https://argoproj.github.io/argo-cd/user-guide/resource_hooks/#using-a-hook-to-send-a-slack-message).

I'm not sure we'll also want to deploy Tekton, but I will look into it more as I've heard of it but am unfamiliar with its capabilities.

Thank you for your response!
